File: frontend/src/lib/api/roi.ts
// =============================================
// ROI 分析 API 客戶端
// =============================================

import { apiClient } from './client'

// ==================== Types ====================

export interface ROISummary {
  total_value_generated: number
  ai_pricing_contribution: number
  competitor_monitoring_value: number
  risk_avoidance_value: number
  roi_percentage: number
  period_start: string
  period_end: string
}

export interface ROITrendPoint {
  date: string
  cumulative_value: number
  ai_pricing: number
  monitoring: number
  risk_avoidance: number
}

export interface ROITrendsResponse {
  trends: ROITrendPoint[]
  start_date: string
  end_date: string
  granularity: string
}

export interface PricingProposalImpact {
  id: string
  product_name: string
  old_price: number
  new_price: number
  price_diff: number
  impact: number
  executed_at: string
}

export interface PricingImpactSummary {
  total_proposals: number
  executed_count: number
  approved_count: number
  rejected_count: number
  total_impact: number
}

export interface PricingImpactResponse {
  proposals: PricingProposalImpact[]
  summary: PricingImpactSummary
}

export interface CompetitorInsights {
  price_alerts_triggered: number
  price_drops_detected: number
  price_increases_detected: number
  avg_response_time_hours: number | null
  potential_savings: number
  period: string
}

export type ROIPeriod = 'today' | 'week' | 'month' | 'quarter'

// ==================== API Functions ====================

/**
 * 獲取 ROI 總覽
 */
export async function getROISummary(period: ROIPeriod = 'month'): Promise<ROISummary> {
  const response = await apiClient.get('/roi/summary', {
    params: { period }
  })
  return response as unknown as ROISummary
}

/**
 * 獲取 ROI 趨勢數據
 */
export async function getROITrends(
  days: number = 30,
  granularity: 'day' | 'week' | 'month' = 'day'
): Promise<ROITrendsResponse> {
  const response = await apiClient.get('/roi/trends', {
    params: { days, granularity }
  })
  return response as unknown as ROITrendsResponse
}

/**
 * 獲取 AI 改價影響分析
 */
export async function getPricingImpact(limit: number = 10): Promise<PricingImpactResponse> {
  const response = await apiClient.get('/roi/pricing-impact', {
    params: { limit }
  })
  return response as unknown as PricingImpactResponse
}

/**
 * 獲取競品監測洞察
 */
export async function getCompetitorInsights(period: ROIPeriod = 'month'): Promise<CompetitorInsights> {
  const response = await apiClient.get('/roi/competitor-insights', {
    params: { period }
  })
  return response as unknown as CompetitorInsights
}

File: backend/app/schemas/roi.py
# =============================================
# ROI 分析 Schema
# =============================================

from typing import List, Optional
from datetime import datetime
from decimal import Decimal
from pydantic import BaseModel, Field


class ROISummary(BaseModel):
    """ROI 總覽數據"""
    total_value_generated: Decimal = Field(..., description="總價值創造 (HKD)")
    ai_pricing_contribution: Decimal = Field(..., description="AI 定價貢獻 (HKD)")
    competitor_monitoring_value: Decimal = Field(..., description="競品監測價值 (HKD)")
    risk_avoidance_value: Decimal = Field(..., description="風險規避價值 (HKD)")
    roi_percentage: float = Field(..., description="投資回報率 (%)")
    period_start: datetime
    period_end: datetime


class ROITrendPoint(BaseModel):
    """ROI 趨勢數據點"""
    date: str = Field(..., description="日期 (YYYY-MM-DD)")
    cumulative_value: Decimal = Field(..., description="累計價值")
    ai_pricing: Decimal = Field(..., description="AI 定價貢獻")
    monitoring: Decimal = Field(..., description="競品監測價值")
    risk_avoidance: Decimal = Field(..., description="風險規避價值")


class ROITrendsResponse(BaseModel):
    """ROI 趨勢響應"""
    trends: List[ROITrendPoint]
    start_date: str
    end_date: str
    granularity: str = "day"


class PricingProposalImpact(BaseModel):
    """AI 改價影響數據"""
    id: str
    product_name: str
    old_price: Decimal
    new_price: Decimal
    price_diff: Decimal = Field(..., description="價格差異")
    impact: Decimal = Field(..., description="影響金額 (價差 × 預估銷量)")
    executed_at: datetime


class PricingImpactSummary(BaseModel):
    """改價影響統計"""
    total_proposals: int
    executed_count: int
    approved_count: int
    rejected_count: int
    total_impact: Decimal


class PricingImpactResponse(BaseModel):
    """改價影響響應"""
    proposals: List[PricingProposalImpact]
    summary: PricingImpactSummary


class CompetitorInsights(BaseModel):
    """競品監測價值洞察"""
    price_alerts_triggered: int = Field(..., description="觸發的價格告警數")
    price_drops_detected: int = Field(..., description="偵測到的降價次數")
    price_increases_detected: int = Field(..., description="偵測到的加價次數")
    avg_response_time_hours: Optional[float] = Field(None, description="平均響應時間 (小時)")
    potential_savings: Decimal = Field(..., description="潛在節省金額")
    period: str
